packages/runtime/container-runtime/src/summarizer.ts

```diff
@@ -143,10 +176,70 @@ export class Summarizer extends EventEmitter implements ISummarizer {
+        const stopReason = await stopP;
+
+        // There are two possible approaches here:
+        // 1. Propagate cancellation from this.stopDeferred to runCoordinator. This will ensure that we move to the exit
+        //    faster, including breaking out of the RunningSummarizer.trySummarize() faster.
+        //    We could create new coordinator and pass it to waitStop() -> trySummarizeOnce("lastSummary") flow.
+        //    The con of this approach is that we might cancel active summary, and lastSummary will fail because it
+        //    did not wait for ack/nack from previous summary. Plus we disregard any 429 kind of info from service
+        //    that way (i.e. trySummarize() loop might have been waiting for 5 min because storage told us so).
+        //    In general, it's more wasted resources.
+        // 2. We can not do it and make waitStop() do last summary only if there was no active summary. This ensures
+        //    that client behaves properly (from server POV) and we do not waste resources. But, it may mean we wait
+        //    substantially longer for trySummarize() retries to play out and thus this summary loop may run into
+        //    conflict with new summarizer client starting on different client.
+        // As of now, #2 is implemented. It's more forward looking, as issue #7279 suggests changing design for new
+        // summarizer client to not be created until current summarizer fully moves to exit, and that would reduce
+        // cons of #2 substantially.
+
+        // Cleanup after running
+        await runningSummarizer.waitStop(!runCoordinator.cancelled /* allowLastSummary */);
+
+        // Propagate reason and ensure that if someone is waiting for cancellation token, they are moving to exit
+        runCoordinator.stop(stopReason);
+
+        return stopReason;
+    }
+
+    /**
+     * Put the summarizer in a started state, including creating and initializing the RunningSummarizer.
+     * The start request can come either from the SummaryManager (in the auto-summarize case) or from the user
+     * (in the on-demand case).
+     * @param onBehalfOf - ID of the client that requested that the summarizer start
+     * @param runCoordinator - cancellation token
+     * @param options - options to forward to the RunningSummarizer
+     * @returns - Promise that is fulfilled when the RunningSummarizer is ready
+     */
+    private async start(
+        onBehalfOf: string,
+        runCoordinator: ICancellableSummarizerController,
+        options?: Readonly<Partial<ISummarizerOptions>>): Promise<RunningSummarizer> {
+        if (this.runningSummarizer) {
+            if (this.runningSummarizer.disposed) {
+                throw new UsageError("Starting a disposed summarizer");
+            }
+            return this.runningSummarizer;
+        }
+        if (this.starting) {
+            throw new UsageError("start() called while summarizer is starting");
```

Comment:
Maybe tune message to what user understands / done? I.e. users do not know what start() refers to here, but they will understand (likely) that certain public API is called twice. It's hard to provide this guidance as we do not know how we got here, but maybe we can generalize but provide message that is more on point to users?

Reply:
Oh, that's true, the user knows nothing about "start".